### DISCUSSION TITLE
unable to extract MoleculeNet dataset

### DISCUSSION THREAD

**Discussion by LanceKnight** [pyg-team/pytorch_geometric#2474] [Q&A]
Hello, 

I downloaded SIDER dataset from MoleculeNet but it requires manual gunzip. The automatic extra of the  .gz file does not work. This makes me unable to use ClearML, which overwrites my manually extracted file everytime.

![image](https://user-images.githubusercontent.com/5760199/116166213-24226700-a6c3-11eb-97ad-4b735f8a1342.png)

**Comment by LanceKnight** [comment]
Also, I cannot find the extract_gz in torch_geometric.data documentation. Can someone help?

**Comment by rusty1s** [comment]
Mh, works for me via:
```python
from torch_geometric.datasets import MoleculeNet
MoleculeNet('/tmp/MoleculeNet', 'Sider')
```
You can find the code of `extract_gz` [here](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/data/extract.py#L51-L55). This is just an internal function, so we do not provide a documentation for it.

**Reply by LanceKnight** [reply]
Hi, @rusty1s ,

I tested with relative and absolute paths. It looks like the extraction does NOT work for the relative path but works for the absolute path.

**Reply by rusty1s** [reply]
You are right. I fixed it in master :) Thanks a lot for catching this.

**Reply by LanceKnight** [reply]
@rusty1s Thank you :)
